pkg/settings/setting.go

```diff
@@ -159,6 +158,10 @@ var (
 
 	// RancherWebhookMinVersion is the minimum version of the webhook that rancher will install
 	RancherWebhookMinVersion = NewSetting("rancher-webhook-min-version", "")
+
+	// SystemDefaultRegistry is the default registry set by users when the environment variable CATTLE_SYSTEM_DEFAULT_REGISTRY
+	// is set. The environment variable CATTLE_BASE_REGISTRY is internal and used to set the base registry as set by SUSE.
+	SystemDefaultRegistry = NewSetting("system-default-registry", os.Getenv("CATTLE_BASE_REGISTRY"))
```

Comment:
This comment is more about Rancher's particular usage of the setting `system-default-registry` and not the variable `SystemDefaultRegistry` itself. I think the information is helpful but is better suited to go into Rancher docs.
> SystemDefaultRegistry is the default registry set by users when the environment variable CATTLE_SYSTEM_DEFAULT_REGISTRY

We can leave this first part off since the fact that this setting is affected by the environmental variable `CATTLE_SYSTEM_DEFAULT_REGISTRY` is an implementation detail of the dashboardapi controller, which is independent of this package.

> The environment variable CATTLE_BASE_REGISTRY is internal and used to set the base registry as set by SUSE.

Our internal usage of the environmental variable `CATTLE_BASE_REGISTRY` is also independent of this package. 
```suggestion
	// SystemDefaultRegistry is the default contrainer registry used for images.
	// The environmental variable "CATTLE_BASE_REGISTRY" controls the default value of this setting.
	SystemDefaultRegistry = NewSetting("system-default-registry", os.Getenv("CATTLE_BASE_REGISTRY"))
```

Reply:
I am looking to add more code comments to the internals of Rancher. The ratio of comments to code is too low and better explanations within the code will be useful in the future. Thanks for helping to clarify that one.

Reply:
Agreed, I added comments to this file earlier this year for that very reason.